### ISSUE TITLE
Удвоение циклов.

### ISSUE THREAD

**Issue by Siorinex** [sergejey/majordomo#281]
После апдейта от 26 августа засек, что в xray-Сервисах некоторые циклы в двух экземплярах. Логику честно не понял...

Удвоены и не запущены:
cycle_asterisk, cycle_nut, cycle_skype, cycle_X101_skype

Удвоены и запущены:
cycle_execs, cycle_main, cycle_mqtt, cycle_phistory, cycle_ping, cycle_scheduler, cycle_states, cycle_sysinfo, cycle_syswidget, cycle_sys_date_format, cycle_websockets, cycle_webvars, cycle_wscache, cycle_xiaomihome, cycle_yandexweather, cycle_zoneminder

В странном состоянии: 
cycle_connect и cycle_connect_manual - незапущены.

В единичном и не запущены:
cycle_modbus, cycle_rss, cycle_watchfolders, cycle_zwave

В единичном и запущен:
cycle_telegram

У запущенных двойняшек - разные времена активации.

**Comment by sergejey** [comment]
Здравствуйте! После перезагрузки системы такая же картина?

**Comment by Siorinex** [comment]
Угу. Как увидел это чудо - сразу и перегрузил...

**Comment by sergejey** [comment]
а в каком окружении работает система? ОС, способ запуска и т.п.

**Comment by Siorinex** [comment]
Ubuntu 18.04
Виртуальная машина на сервере: там кроме Мажордомо и его базы нет даже MQTT - оно на другой виртуалке.
Автозапуск и т.д. и т.п.

Вообще, подозреваю хвосты появились ещё с начального этапа установки: помнишь ты ещё на Телеграмме посоветовал снести базу и установить от Альфа-ветки? Все заработало, но после установки модуля датчика Температура-Влажность я засёк дублирование классов. Причём оно как-то именно дублировало именно классы - объекты были в единичном экземпляре и только в одном из классов.

АФЗ, честно. Особенно с учетом того, что некоторые модули от которых висит я и вообще не ставил... Телега, которую ставил буквально чуть не позавчера - в единичном. Как-то даже раньше в этот раздел не залезал, а тут внезапно всплыло...

**Comment by sergejey** [comment]
ок... а если в консоли набрать ps aux|grep php
что будет?
у меня всё же подозрение, что как-то не нормально настроен авто-запуск и идёт дублирование циклов, а оно уже пораждает другие глюки

**Comment by Siorinex** [comment]
root       333  0.1  2.7 307796 27660 ?        Ss   Aug26   3:33 /usr/bin/php /v                                                                                        ar/www/cycle.php >/dev/nul
root       481  0.0  2.7 303696 28168 ?        S    Aug26   1:11 php -q ./script                                                                                        s/cycle_execs.php --params a:0:{}
root       482  0.1  2.7 303696 28096 ?        S    Aug26   4:09 php -q ./script                                                                                        s/cycle_wscache.php --params a:0:{}
root       485  0.1  2.8 308192 29232 ?        S    Aug26   3:44 php -q ./script                                                                                        s/cycle_syswidget.php --params a:0:{}
root       486  0.1  2.7 303696 28204 ?        S    Aug26   3:37 php -q ./script                                                                                        s/cycle_sysinfo.php --params a:0:{}
root       488  0.0  2.8 383684 28680 ?        S    Aug26   0:23 php -q ./script                                                                                        s/cycle_webvars.php --params a:0:{}
root       489  0.1  2.7 303696 27952 ?        S    Aug26   3:55 php -q ./script                                                                                        s/cycle_sys_date_format.php --params a:0:{}
root       490  0.1  2.7 303696 28100 ?        S    Aug26   3:48 php -q ./script                                                                                        s/cycle_scheduler.php --params a:0:{}
root       491  0.0  2.8 303696 28452 ?        S    Aug26   2:09 php -q ./script                                                                                        s/cycle_mqtt.php --params a:0:{}
root       492  0.1  2.9 382556 29992 ?        S    Aug26   4:12 php -q ./script                                                                                        s/cycle_telegram.php --params a:0:{}
root       495  0.0  2.9 383952 29984 ?        S    Aug26   1:16 php -q ./script                                                                                        s/cycle_main.php --params a:0:{}
root       496  0.0  2.8 307864 28344 ?        S    Aug26   0:22 php -q ./script                                                                                        s/cycle_connect.php --params a:0:{}
root       498  0.1  2.7 303696 28108 ?        S    Aug26   2:21 php -q ./script                                                                                        s/cycle_phistory.php --params a:0:{}
root       499  0.2  2.8 303696 28428 ?        S    Aug26   4:45 php -q ./script                                                                                        s/cycle_states.php --params a:0:{}
root       506  0.0  2.7 303696 28216 ?        S    Aug26   1:25 php -q ./script                                                                                        s/cycle_ping.php --params a:0:{}
root     16546  0.9  2.8 303696 28652 ?        S    10:12   0:16 php -q ./script                                                                                        s/cycle_websockets.php --params a:0:{}
pawel    17241  0.0  0.1  14428  1092 pts/0    S+   10:40   0:00 grep --color=au                                                                                        to php
root     18015  0.1  2.8 307892 28704 ?        S    Aug27   1:58 php -q ./script                                                                                        s/cycle_zoneminder.php --params a:0:{}
root     18023  0.2  2.9 308032 29292 ?        S    Aug27   3:32 php -q ./script                                                                                        s/cycle_yandexweather.php --params a:0:{}
root     19900  0.0  2.8 303696 28412 ?        S    Aug27   0:51 php -q ./script                                                                                        s/cycle_xiaomihome.php --params a:0:{}

**Comment by sergejey** [comment]
хм... вроде и не дублируется ничего :-/ странно тогда. буду думать...

**Comment by Siorinex** [comment]
Вообще на уровне бреда: а X-Ray откуда берёт инфу? Это табличка в базе или что?

**Comment by sergejey** [comment]
Да, берутся все свойства, которые по названию подпадают под паттерн состояния цикла и выводятся. Сейчас (в альфа-ветке) при старте системы удаляются все такие свойства, чтобы циклы могли сами их создать заново. Попробуйте пожалуйста и отпишите, как оно себя ведёт с последними обновлениями.

**Comment by Siorinex** [comment]
Последний - "Market module update". Ресет 100% был - я его ручками делаю, т.к. иначе часы на терминалах встают...

Все то-же, все те-же. Только теперь и телеграмм двоится...
